### PR TITLE
Add scheduler attribute for checks

### LINKED ISSUE #2891: Document new scheduler attribute for checks
### Description
Add the new scheduler attribute to the check reference and API doc.

### Category

This is an issue with:

- [ ] errata (error in doc content)
- [ ] new doc (requires a completely new doc)
- [x] update (add missing or refresh existing content)
- [ ] improvement (improve docs content or format)

### Affected pages
https://docs.sensu.io/sensu-go/latest/observability-pipeline/observe-schedule/checks/
https://docs.sensu.io/sensu-go/latest/api/checks/

### Expected behavior
Docs list `scheduler` in returned check examples. Check spec includes a description table for the new `scheduler` attribute.

### Current behavior
Not currently documented -- to be released in 6.2.0

### Corrections needed
Add `scheduler` to returned check examples and check spec.

### Additional context
https://github.com/sensu/sensu-go/pull/4089

### PR DESCRIPTION
**Pull request by hillaryfraley**
## Description
- Adds scheduler attribute description table in the check reference
- Adds scheduler attribute to GET examples in the checks API doc

## Motivation and Context
Closes https://github.com/sensu/sensu-docs/issues/2891

### CODE REVIEW

**Comment by hillaryfraley** [comment]
@echlebek Do we need to mention anything about this in https://docs.sensu.io/sensu-go/latest/observability-pipeline/observe-schedule/checks/#check-scheduling?

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/checks.md [hunk 1]
```diff
@@ -720,6 +720,13 @@ required     | false
 type         | Array
 example      | {{< code shell >}}"env_vars": ["RUBY_VERSION=2.5.0", "CHECK_HOST=my.host.internal"]{{< /code >}}
 
+|scheduler  |     |
```
Comment (hillaryfraley) [inline content/sensu-go/6.2/observability-pipeline/observe-schedule/checks.md:733]:
```suggestion
<a name="scheduler-attribute"></a>

|scheduler  |     |
```

File: content/sensu-go/6.2/observability-pipeline/observe-schedule/checks.md [hunk 2]
```diff
@@ -720,6 +720,15 @@ required     | false
 type         | Array
 example      | {{< code shell >}}"env_vars": ["RUBY_VERSION=2.5.0", "CHECK_HOST=my.host.internal"]{{< /code >}}
 
+<a name="scheduler-attribute"></a>
+
+|scheduler  |     |
+------------|-----
+description | Type of scheduler that schedules the check. Sensu automatically sets the `scheduler` value and overrides any user-entered values. May be `memory` for checks scheduled in-memory, `etcd` for checks scheduled with etcd leases and watchers, or `postgres` for checks scheduled with PostgreSQL using transactions and asynchronous notification (NOTIFY).
```
Comment (cwjohnston) [inline content/sensu-go/6.2/observability-pipeline/observe-schedule/checks.md]: > Sensu automatically sets the `scheduler` value

I'm inferring this is automatic based on settings like PostgresConfig `enable_round_robin`. If that is correct, can we explain the connection here, or in PostgresConfig docs?

Comment (hillaryfraley) [reply content/sensu-go/6.2/observability-pipeline/observe-schedule/checks.md]: I added some explanation based on the same inference, but I couldn't find anything in the sensu-go PR to confirm. @echlebek can you double-check the explanation I added here and make sure I didn't misrepresent how Sensu derives the scheduler values?

**Comment by hillaryfraley** [comment]
Eric :ok: via Slack https://sensu.slack.com/archives/CCA1088MC/p1608231021310600

**Pull request merged**